src/DnsClient/IDnsQuery.cs

```diff
@@ -47,6 +47,29 @@ public interface IDnsQuery
         /// <exception cref="DnsResponseException">After retries and fallbacks, if none of the servers were accessible, timed out or (if <see cref="DnsQueryOptions.ThrowDnsErrors"/> is enabled) returned error results.</exception>
         IDnsQueryResponse Query(DnsQuestion question, DnsQueryAndServerOptions queryOptions);
 
+        /// <summary>
+        /// Performs a lookup for the given <paramref name="question" /> against the in-memory cache.

```

Comment:
Maybe change the description of those 2 methods to.
`Returns cached results for the given  <paramref name="question" /> from the in-memory cache, if available, or <c>Null</c> otherwise.`

And as Remarks:
`This method will not perform a full lookup if there is nothing found in cache or the cache is disabled!`